bundles/org.openhab.binding.flume/src/main/java/org/openhab/binding/flume/internal/handler/FlumeBridgeHandler.java

```diff
@@ -0,0 +1,286 @@
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FlumeBridgeHandler} implements the Flume bridge cloud connector
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class FlumeBridgeHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(FlumeBridgeHandler.class);
+
+    public FlumeBridgeConfig config = new FlumeBridgeConfig();
+
+    private static final Duration CACHE_EXPIRY = Duration.ofMinutes(30);
+    private ExpiringCache<List<FlumeApiDevice>> apiListDevicesCache = new ExpiringCache<>(CACHE_EXPIRY,
+            this::apiListDevicesAction);
+
+    private boolean logOnce = false;
+
+    private final FlumeApi api;
+    final SystemOfUnits systemOfUnits;
+    final TranslationProvider i18nProvider;
+    final LocaleProvider localeProvider;
+    final Bundle bundle;
+
+    public FlumeApi getApi() {
+        return api;
+    }
+
+    protected @Nullable ScheduledFuture<?> pollingJob;
+    private @Nullable FlumeDiscoveryService discoveryService;
+
+    /**
+     * Get the services registered for this bridge. Provides the discovery service.
+     */
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(FlumeDiscoveryService.class);
+    }
+
+    public boolean registerDiscoveryListener(FlumeDiscoveryService listener) {
+        if (discoveryService == null) {
+            discoveryService = listener;
+            return true;
+        }
+
+        return false;
+    }
+
+    public boolean unregisterDiscoveryListener() {
+        if (discoveryService != null) {
+            discoveryService = null;
+            return true;
+        }
+
+        return false;
+    }
+
+    public FlumeBridgeHandler(final Bridge bridge, SystemOfUnits systemOfUnits, HttpClient httpClient,
+            TranslationProvider i18nProvider, LocaleProvider localeProvider) {
+        super(bridge);
+
+        api = new FlumeApi(httpClient);
+        this.systemOfUnits = systemOfUnits;
+        this.i18nProvider = i18nProvider;
+        this.localeProvider = localeProvider;
+        this.bundle = FrameworkUtil.getBundle(this.getClass());
+    }
+
+    public FlumeBridgeConfig getFlumeBridgeConfig() {
+        return config;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(FlumeBridgeConfig.class);
+
+        if (config.clientId.isBlank() | config.clientSecret.isBlank() || config.password.isBlank()
+                || config.username.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "@text/offline.cloud-configuration-error");
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        goOnline();
```

Comment:
Initilize should return as fast as possible. `goOnline()` calls => `api.initialize()` calls => `getToken()` => performing a http request (and thus blocking) 
A simple solution would be use the scheduler: 

```suggestion
        scheduler.schedule(this::goOnline, 0, TimeUnit.SECONDS);
```

Reply:
@lsiepel, does scheduler.execute(this::goOnline) accomplish the same thing?  Just curious the difference between the two calls?